attributes/default.rb

```diff
@@ -26,6 +26,6 @@
   }
 default['osl-docker']['prune']['volume_filter'] = []
 default['osl-docker']['tls'] = false
-default['osl-docker']['host'] = node['osl-docker']['tls'] ? 'tcp://127.0.0.1:2376' : nil
+default['osl-docker']['host'] = node['osl-docker']['tls'] ? 'tcp://127.0.0.1:2376' : ''
```

Comment:
Why was this changed?

Reply:
The new `osl-shell` cannot have `nil` as an argument, and even though theres a `not_if` guard it still throws a validation error. I changed the default value attribute value and the nil checks to check for empty string instead.

Reply:
So it looks like to properly work around this we should change ``osl_shell_environment[DOCKER_HOST]`` to be wrapped in an ``if`` block instead of using the ``not_if`` guard. That should work around the issue properly and not require changing this.